mypy/config_parser.py

```diff
@@ -63,8 +71,8 @@ def split_and_match_files(paths: str) -> List[str]:
     'python_version': parse_version,
     'strict_optional_whitelist': lambda s: s.split(),
     'custom_typing_module': str,
-    'custom_typeshed_dir': str,
-    'mypy_path': lambda s: [p.strip() for p in re.split('[,:]', s)],
+    'custom_typeshed_dir': expand_path,
+    'mypy_path': lambda s: [expand_path(p.strip()) for p in re.split('[,:]', s)],
     'files': split_and_match_files,
     'quickstart_file': str,
     'junit_xml': str,
```

Comment:
Would enabling environment variable interpolation for this config key be as simple as this?
```suggestion
    'junit_xml': expand_path
```

Reply:
> Would enabling environment variable interpolation for this config key be as simple as this?

Yes. Probably should have done this for `junit_xml` and `quickstart_file` as well.

Reply:
Alright, I'll include that in my PR #8479